src/Playroom/Toolbar/Toolbar.tsx

```diff
@@ -148,58 +154,65 @@ export default ({ themes: allThemes, widths: allWidths, snippets }: Props) => {
             </ToolbarItem>
           </div>
         </div>
-
-        <div className={styles.panel}>
-          {isSnippetsOpen && (
-            <div
-              hidden={isSnippetsOpen ? undefined : true}
-              className={styles.preference}
-            >
-              <Snippets
-                snippets={snippets}
-                onHighlight={(snippet) => {
-                  dispatch({
-                    type: 'previewSnippet',
-                    payload: { snippet },
-                  });
-                }}
-                onClose={(snippet) => {
-                  if (snippet) {
+        <CSSTransition
+          in={isOpen}
+          timeout={ANIMATION_TIMEOUT}
+          classNames={styles.transitionStyles}
+          mountOnEnter
+          unmountOnExit
+          onEnter={() => setLastActivePanel(activeToolbarPanel)}
+          onExited={() => setLastActivePanel(undefined)}
+        >
+          <div className={styles.panel} id="custom-id">
+            {lastActivePanel === 'snippets' && (
+              <div className={styles.preference}>
```

Comment:
All of the conditionally rendered panels share a top-level `div` with just a `styles.preference` classname. Could the `div` maybe be hoisted outside the conditional render, and hence deduplicated, or maybe conditionally applied alongside `styles.panel` on line 166?

Reply:
Good idea, I reckon just wrapping it around all the conditional renders is probably the simplest option.